### PR TITLE
Add Immix GC

### PR DESCRIPTION
**Pull request by wenyuzhao**

### CODE REVIEW

File: mmtk/src/vm_metadata/constants.rs [hunk 1]
```diff
@@ -35,19 +31,18 @@ pub(crate) const FORWARDING_BITS_METADATA_SPEC: VMLocalForwardingBitsSpec =
 /// 1 bit per object
 #[cfg(feature = "mark_bit_in_header")]
 pub(crate) const MARKING_METADATA_SPEC: VMLocalMarkBitSpec =
-    VMLocalMarkBitSpec::in_header(FORWARDING_BITS_OFFSET);
+    VMLocalMarkBitSpec::side(match LOS_METADATA_SPEC.get_spec() {
```
Comment (qinsoon) [inline mmtk/src/vm_metadata/constants.rs:34]: This needs to be updated to the new API.

**Review by qinsoon — COMMENTED** [review]
Please add tests for immix in ci-test.

File: .github/scripts/ci-test.sh [hunk 2]
```diff
@@ -116,3 +116,21 @@ build/linux-x86_64-normal-server-$DEBUG_LEVEL/jdk/bin/java -XX:+UseThirdPartyHea
 build/linux-x86_64-normal-server-$DEBUG_LEVEL/jdk/bin/java -XX:+UseThirdPartyHeap -server -XX:MetaspaceSize=100M -Xms4G -Xmx4G -jar benchmarks/dacapo-2006-10-MR2.jar fop
 build/linux-x86_64-normal-server-$DEBUG_LEVEL/jdk/bin/java -XX:+UseThirdPartyHeap -server -XX:MetaspaceSize=100M -Xms4G -Xmx4G -jar benchmarks/dacapo-2006-10-MR2.jar luindex
 # build/linux-x86_64-normal-server-$DEBUG_LEVEL/jdk/bin/java -XX:+UseThirdPartyHeap -server -XX:MetaspaceSize=100M -Xms4G -Xmx4G -jar benchmarks/dacapo-2006-10-MR2.jar pmd
+
+# --- Immix ---
+export MMTK_PLAN=Immix
```
Comment (qinsoon) [inline .github/scripts/ci-test.sh:121]: Can you move the tests above the PageProtect tests? The page protect changes the max_map_count - we should not run immix with increased max_map_count.

**Comment by qinsoon** [comment]
The test also needs a longer timeout. Now the timeout is 120min.

File: mmtk/src/vm_metadata/constants.rs (hunk 1 above)
Comment (wenyuzhao) [reply mmtk/src/vm_metadata/constants.rs]: Fixed

File: .github/scripts/ci-test.sh (hunk 2 above)
Comment (wenyuzhao) [reply .github/scripts/ci-test.sh]: Fixed

**Review by qinsoon — APPROVED** [review]

**Pull request merged**